core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -503,9 +503,25 @@ class CityConstructions {
         if (!cityInfo.civInfo.gameInfo.gameParameters.godMode) {
             val construction = getConstruction(constructionName)
             if (construction is PerpetualConstruction) return false
-            val constructionCost = (construction as INonPerpetualConstruction).getStatBuyCost(cityInfo, stat)
+            val constructionCost =
+                (construction as INonPerpetualConstruction).getStatBuyCost(cityInfo, stat)
             if (constructionCost == null) return false // We should never end up here anyway, so things have already gone _way_ wrong
             cityInfo.addStat(stat, -1 * constructionCost)
+
+            if (cityInfo.civInfo.getMatchingUniques("May buy [] units for [] [] [] starting from the [] at an increasing price ([])")
+                .any {
+                    (
+                        construction is BaseUnit && construction.matchesFilter(it.params[0]) ||
+                        construction is Building && construction.matchesFilter(it.params[0])
```

Comment:
Unique text says "[] units" - support `Building` anyway? And - the newbie-puzzling `||` wouldn't be necessary if `matchesFilter` were part of `INonPerpetualConstruction`, right?